docs/fundamentals/persist-data-volumes.md

```diff
@@ -26,39 +26,71 @@ Volumes are the recommended way to persist data generated by containers and supp
+
+:::code language="csharp" source="snippets/volumes/VolumeMounts.AppHost/Program.WithBindMount.cs" id="mount":::
+
 In this example:
 
-- `VolumeMount.sqlserver.data` sets where the volume will be stored on the host.
+- `VolumeMount.AppHost-sql-data` sets where the volume will be stored on the host.
 - `/var/opt/mssql` sets the path to the database files in the container.
 
+All .NET Aspire container resources can utilize volume mounts, and some provide convenient APIs for adding named volumes derived from resources. Using the `WithDataVolume` as an example, the following code is functionally equivalent to the previous example but more succinct:
+
+:::code language="csharp" source="snippets/volumes/VolumeMounts.AppHost/Program.Implicit.cs" id="implicit":::
+
+With the app host project being named `VolumeMount.AppHost`, the `WithDataVolume` method automatically creates a named volume as `VolumeMount.AppHost-sql-data` and is mounted to the `/var/opt/mssql` path in the SQL Server container. The naming convention is as follows:
+
+- `{appHostProjectName}-{resourceName}-data`: The volume name is derived from the app host project name and the resource name.
+
 ## Create a persistent password
 
-Named volumes require a consistent password between app launches. Run the following command in your project directory to set a local password in your .NET user secrets:
+Named volumes require a consistent password between app launches. .NET Aspire conveniently provides random password generation functionality. Consider the previous example once more, where a password is generated automatically:
+
+:::code language="csharp" source="snippets/volumes/VolumeMounts.AppHost/Program.Implicit.cs" id="implicit":::
+
+Since the `password` parameter isn't provided when calling `AddSqlServer`, .NET Aspire automatically generates a password for the SQL Server resource.
+
+> [!IMPORTANT]
+> This isn't a persistent password! Instead, it changes every time the app host runs.
+
+To create a _persistent_ password, you must override the generated password. To do this, run the following command in your app host project directory to set a local password in your .NET user secrets:
 
 ```dotnetcli
-dotnet user-secrets set samplepassword <password>
+dotnet user-secrets set Parameters:sql-password <password>
 ```
 
-Retrieve this password in your app using the following code:
+The naming convention for these secrets is important to understand. The password is stored in configuration with the `Parameters:sql-password` key. The naming convention follows this pattern:
 
-```csharp
-// Retrieve the password you created from user secrets
-var password = builder.Configuration["samplepassword"];
-```
+- `Parameters:{resourceName}-password`: In the case of the SQL Server resource (which was named `"sql"`), the password is stored in the configuration with the key `Parameters:sql-password`.
```

Comment:
@IEvangelist to avoid conflicts between volumes from different app hosts, the volume names are prefixed with the app host name. Wouldn't it make sense to do the same for parameters?